aws/table_aws_batch_job.go

```diff
@@ -0,0 +1,331 @@
+			},
+			{
+				Name:        "timeout",
+				Description: "The timeout configuration for the job.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getBatchJob,
+			},
+
+			// Steampipe standard columns
+			{
+				Name:        "title",
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("JobName"),
+			},
+			{
+				Name:        "tags",
+				Description: resourceInterfaceDescription("tags"),
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getBatchJob,
+				Transform:   transform.FromValue(),
+			},
+			{
+				Name:        "akas",
+				Description: resourceInterfaceDescription("akas"),
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("JobArn").Transform(transform.EnsureStringArray),
+			},
+		}),
+	}
+}
+
+//// LIST FUNCTION
+
+func listAwsBatchJobs(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	// Create session
+	svc, err := BatchClient(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_batch_job.listAwsBatchJobs", "connection_error", err)
+		return nil, err
+	}
+	if svc == nil {
+		// Unsupported region, return no data
+		return nil, nil
+	}
+
+	// Limiting the results
+	maxLimit := int32(100)
+	if d.QueryContext.Limit != nil {
+		limit := int32(*d.QueryContext.Limit)
+		if limit < maxLimit {
+			maxLimit = limit
+		}
+	}
+
+	input := &batch.ListJobsInput{
+		MaxResults: aws.Int32(maxLimit),
+	}
+
+	paginator := batch.NewListJobsPaginator(svc, input, func(o *batch.ListJobsPaginatorOptions) {
+		o.Limit = maxLimit
+		o.StopOnDuplicateToken = true
+	})
+
+	// List call
+	for paginator.HasMorePages() {
+		// apply rate limiting
+		d.WaitForListRateLimit(ctx)
+
+		output, err := paginator.NextPage(ctx)
+		if err != nil {
+			plugin.Logger(ctx).Error("aws_batch_job.listAwsBatchJobs", "api_error", err)
+			return nil, err
+		}
+
+		for _, items := range output.JobSummaryList {
+			d.StreamListItem(ctx, items)
+
+			// Context can be cancelled due to manual cancellation or the limit has been hit
+			if d.RowsRemaining(ctx) == 0 {
+				return nil, nil
+			}
+		}
+	}
+	return nil, nil
+}
+
+//// HYDRATE FUNCTIONS
+
+func getBatchJob(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+
+	jobId := d.EqualsQualString("id")
+	if jobId == "" {
+		return nil, nil
+	}
+
+	// Create Session
+	svc, err := BatchClient(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_batch_job.getBatchJob", "connection_error", err)
+		return nil, err
+	}
+	if svc == nil {
+		// Unsupported region, return no data
+		return nil, nil
+	}
+
+	// Build the params
+	params := &batch.DescribeJobsInput{
+		Jobs: []string{jobId},
+	}
+
+	// Get call
+	data, err := svc.DescribeJobs(ctx, params)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_batch_job.getBatchJob", "api_error", err)
+		return nil, err
+	}
+
+	return data.Jobs, nil
```

Comment:
We should return to a single Job instead of an array of Jobs. The [DescripeJobs](https://docs.aws.amazon.com/batch/latest/APIReference/API_DescribeJobs.html) API returns a list of jobs as a response.